projects/ngx-prx-styleguide/src/lib/icon/icon.component.ts

```diff
@@ -38,24 +43,7 @@ export class IconComponent implements OnChanges {
   @HostBinding('style.width') width: string;
   @HostBinding('style.height') height: string;
 
-  private hostClasses: { [name: string]: boolean };
-
-  constructor(@Self() protected ngClass: NgClass) { }
-
-  ngOnChanges(): void {
-    this.hostClasses = {};
-
-    if (!!this._color) {
-      this.hostClasses[`color--${this._color}`] = true;
-    }
-
-    this.updateHostClasses();
-  }
-
-  private updateHostClasses(): void {
-    this.ngClass.ngClass = this.hostClasses;
-    this.ngClass.ngDoCheck();
-  }
+  constructor(private renderer: Renderer2, private el: ElementRef) {}
```

Comment:
Angular 8 doesn't support injecting NgClass into a component, but it does let you set/add/remove the host class with Renderer2